File: fizzbuzz.rb
class FizzBuzzGenerator
  def translate(list)
    list.map do |number|
      if number % 3 == 0 && number % 5 == 0
        number = 'fizzbuzz'
      elsif number % 3 == 0
        number = 'fizz'
      elsif number % 5 == 0
        number = 'buzz'
      else
        number
      end
    end
  end
end

File: fizzbuzz.nim
import strutils

proc fizzbuzz*(number: int): string =
  if number %% 15 == 0:
    "fizzbuzz"
  elif number %% 3 == 0:
    "fizz"
  elif number %% 5 == 0:
    "buzz"
  else:
    intToStr(number)
